analyzer/src/main/kotlin/YamlFilePackageCurationProvider.kt

```diff
@@ -35,5 +38,21 @@ class YamlFilePackageCurationProvider(
         curationFile.readValue<List<PackageCuration>>()
     }
 
-    override fun getCurationsFor(identifier: Identifier) = packageCurations.filter { it.id.matches(identifier) }
+    override fun getCurationsFor(packageId: Identifier): List<PackageCuration> {
+        val packageVersion = try {
+            Semver(packageId.version, Semver.SemverType.IVY)
+        } catch (e: SemverException) {
+            null
+        }
+
+        return packageCurations.filter { (curationId, _) ->
+            val resolvedVersion = try {
+                if (packageVersion?.satisfies(curationId.version) == true) packageId.version else curationId.version
```

Comment:
Is try/catch needed? (satisfies seems not to throw)

Reply:
I believe throwing of `SemverException` could happen deep inside `buildIvy()` which is called by `satisfies()`.

Reply:
I wanted to construct as many objects as possible (read: the `packageVersion`) outside of the inner loop.

Reply:
It seems a bit wasteful to run this comparison for every single available curation, even if other properties do not match. Any reason not to move this code into `Identifier.matches()` and run it after checking `type`, `namespace`, and `name`?

Reply:
I simply wasn't sure whether we always want to allow version wildcards, or only in the context of curations. Should we maybe add an optional boolean parameter to `Identifier.matches()`, defaulting to false, whether to allow version ranges? Or should we simply always support them?